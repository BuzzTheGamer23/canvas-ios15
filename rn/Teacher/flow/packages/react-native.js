// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
