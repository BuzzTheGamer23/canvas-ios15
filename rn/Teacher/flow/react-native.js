//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
