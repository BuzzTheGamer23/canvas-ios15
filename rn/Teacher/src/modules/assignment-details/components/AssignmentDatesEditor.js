/**
 * @flow
 */

import React, { Component } from 'react'

import {
  View,
  Text,
  TouchableHighlight,
  StyleSheet,
} from 'react-native'
import i18n from 'format-message'
import colors from '../../../common/colors'
import { formattedDueDate } from '../../../common/formatters'
import AssignmentDates from '../../../common/AssignmentDates'

type Props = {
  assignment: Assignment,
}

export default class AssignmentDatesEditor extends Component<any, Props, any> {

  renderDate = (date: AssignmentDate): React.Element<View> => {
    const dateFormatter = (stringDate: ?string): string => {
      return stringDate ? formattedDueDate(new Date(stringDate)) : '--'
    }

    let title = date.title
    if (date.base) {
      const dates = new AssignmentDates(this.props.assignment)
      if (dates.allDates().length > 1) {
        title = i18n('Everyone else')
      } else {
        title = i18n('Everyone')
      }
    }

    return (<View style={styles.dateContainer} key={date.id || 'base'}>
              <View style={styles.row}>
                <Text style={styles.titleText}>{i18n('Assignees')}</Text>
                <Text style={styles.detailText}>{title}</Text>
              </View>
              <View style={styles.row}>
                <Text style={styles.titleText}>{i18n('Due Date')}</Text>
                <Text style={styles.detailText}>{dateFormatter(date.due_at)}</Text>
              </View>
              <View style={styles.row}>
                <Text style={styles.titleText}>{i18n('Available From')}</Text>
                <Text style={styles.detailText}>{dateFormatter(date.unlock_at)}</Text>
              </View>
              <View style={styles.row}>
                <Text style={styles.titleText}>{i18n('Available To')}</Text>
                <Text style={styles.detailText}>{dateFormatter(date.lock_at)}</Text>
              </View>
              <View style={styles.space} />
            </View>)
  }

  renderButton = (): React.Element<View> => {
    return (<TouchableHighlight style={styles.button}>
              <View style={styles.buttonInnerContainer}>
                <Text style={styles.buttonText}>{i18n('Add Due Date')}</Text>
              </View>
            </TouchableHighlight>)
  }

  render (): React.Element<View> {
    const dates = new AssignmentDates(this.props.assignment)
    const rows = dates.allDates().map(this.renderDate)
    const button = this.renderButton()
    return (<View style={styles.container}>
              {rows}
              {button}
              <View style={styles.space} />
            </View>)
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  dateContainer: {

  },
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: 54,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: colors.seperatorColor,
    paddingLeft: global.style.defaultPadding,
    paddingRight: global.style.defaultPadding,
  },
  titleText: {
    fontSize: 16,
    fontWeight: 'bold',
    color: colors.darkText,
  },
  detailText: {
    fontSize: 16,
    fontWeight: '500',
    color: '#008EE2',
  },
  space: {
    height: 24,
    backgroundColor: '#F5F5F5',
  },
  button: {
    height: 54,
  },
  buttonInnerContainer: {
    flex: 1,
    backgroundColor: 'white',
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: global.style.defaultPadding,
    paddingRight: global.style.defaultPadding,
  },
  buttonText: {
    fontSize: 16,
    fontWeight: 'bold',
    color: '#008EE2',
  },
})
