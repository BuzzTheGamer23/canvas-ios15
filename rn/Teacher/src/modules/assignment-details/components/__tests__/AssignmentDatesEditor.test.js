/**
 * @flow
 */

import 'react-native'
import React from 'react'
import AssignmentDatesEditor from '../AssignmentDatesEditor'
import renderer from 'react-test-renderer'

const template = {
  ...require('../../../../api/canvas-api/__templates__/assignments'),
}

test('render with multiple due dates', () => {
  const assignment = template.assignment({
    all_dates: [template.assignmentDueDate({ base: true }), template.assignmentDueDate({ base: false })],
  })
  let tree = renderer.create(
    <AssignmentDatesEditor assignment={assignment} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render with single due date', () => {
  const assignment = template.assignment({
    all_dates: [template.assignmentDueDate({ base: true })],
  })
  let tree = renderer.create(
    <AssignmentDatesEditor assignment={assignment} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
