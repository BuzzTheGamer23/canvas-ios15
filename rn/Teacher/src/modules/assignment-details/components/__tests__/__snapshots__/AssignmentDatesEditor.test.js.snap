// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots! doing stuff with selecting assignees 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#C7CDD1",
            "borderBottomWidth": 0.5,
            "height": 54,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Assignees
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#C7CDD1",
            "borderBottomWidth": 0.5,
            "height": 54,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Due Date
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          --
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#C7CDD1",
            "borderBottomWidth": 0.5,
            "height": 54,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Available From
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          --
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#C7CDD1",
            "borderBottomWidth": 0.5,
            "height": 54,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Available To
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          --
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F5F5F5",
          "height": 24,
        }
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "height": 54,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#008EE2",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Add Due Date
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "height": 24,
      }
    }
  />
</View>
`;

exports[`snapshots! render with multiple due dates 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#C7CDD1",
            "borderBottomWidth": 0.5,
            "height": 54,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Assignees
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          Everyone else
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#C7CDD1",
            "borderBottomWidth": 0.5,
            "height": 54,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Due Date
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          May 31, 2017 at 11:59 PM
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#C7CDD1",
            "borderBottomWidth": 0.5,
            "height": 54,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Available From
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          Mar 17, 2017 at 12:00 AM
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#C7CDD1",
            "borderBottomWidth": 0.5,
            "height": 54,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Available To
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          May 31, 2017 at 11:59 PM
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F5F5F5",
          "height": 24,
        }
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "height": 54,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#008EE2",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Add Due Date
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "height": 24,
      }
    }
  />
</View>
`;

exports[`snapshots! render with single due date 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#C7CDD1",
            "borderBottomWidth": 0.5,
            "height": 54,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Assignees
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          Everyone
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#C7CDD1",
            "borderBottomWidth": 0.5,
            "height": 54,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Due Date
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          May 31, 2017 at 11:59 PM
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#C7CDD1",
            "borderBottomWidth": 0.5,
            "height": 54,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Available From
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          Mar 17, 2017 at 12:00 AM
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#C7CDD1",
            "borderBottomWidth": 0.5,
            "height": 54,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Available To
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 16,
              "fontWeight": "500",
            }
          }
        >
          May 31, 2017 at 11:59 PM
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F5F5F5",
          "height": 24,
        }
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "height": 54,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#008EE2",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Add Due Date
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "height": 24,
      }
    }
  />
</View>
`;
